src/spline/smooth_spline.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2023 ICHIRO ITS
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include "keisan/spline/smooth_spline.hpp"
+
+#include <vector>
+
+namespace keisan
+{
+
+SmoothSpline::SmoothSpline() {}
+
+SmoothSpline::~SmoothSpline() {}
+
+SmoothSpline::SmoothSpline(const std::vector<PointTime>& point_times)
+{
+  for (auto point_time : point_times) {
+    add_point(point_time.point, point_time.time);
+  }
+}
+
+void SmoothSpline::add_point(const Polynom& polynom) { splines.push_back(polynom); }
```

Comment:
If `splines` is exposed as a public method, there's no need to have this function.